### PR TITLE
Fix `engine` and `knowledge` library versions in `workflow` and `testing`

### PR DESCRIPTION
**Pull request by divyaramnath-13**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
Fix `engine` and `knowledge` library versions in `workflow` and `testing`

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -172,6 +172,12 @@ object Dependencies {
   const val androidFhirEngineModule = "engine"
   const val androidFhirEngine =
     "$androidFhirGroup:$androidFhirEngineModule:${Versions.androidFhirEngine}"
+  const val androidFhirKnowledgeModule = "knowledge"
```
Comment (omarismail94) [inline buildSrc/src/main/kotlin/Dependencies.kt:175]: instead of defining these and their versions (it sort of is getting ugly), can we do this:
```
androidFhirEngine = "${Releases.groupId}:${Releases.Engine.artifactId}:${Releases.Engine.version}"
```
this, way, we don't need to  update the version and all other field in this file, and in Releases

Comment (jingtang10) [reply buildSrc/src/main/kotlin/Dependencies.kt]: release version and dependency version should be kept separate. this is because we do need to release the library on maven before others can depend on it. so you always need 2 PRs

Comment (divyaramnath-13) [reply buildSrc/src/main/kotlin/Dependencies.kt]: Got it

File: testing/build.gradle.kts [hunk 2]
```diff
@@ -16,7 +16,7 @@ android {
 }
 
 dependencies {
-  api(project(":engine"))
+  api(Dependencies.androidFhirEngine)
```
Comment (jingtang10) [inline testing/build.gradle.kts:19]: why this change? is this run normally in local testing environment or no kokoro, in which case, should this depend on the project module on HEAD actually?

**Review by jingtang10 — APPROVED** [review]
@divyaramnath-13 thanks for this - great job!

just one question actually - please address or leave reasoning. either way happy for this to be merged.

File: testing/build.gradle.kts (hunk 2 above)
Comment (divyaramnath-13) [reply testing/build.gradle.kts:19]: This is because `testing` is a dependency in `workflow` and they would depend on different versions of `engine` without this change.
